### ISSUE TITLE
ListJobsAsync() returns only us-central1 jobs

### ISSUE THREAD

**Issue by Rosti81** [googleapis/google-cloud-dotnet#7038]
#### Environment details

  - OS: **Windows 10 Enterprise**
  - .NET version: **.NET Core 3.1**
  - Package name and version: **Google.Cloud.Dataflow.V1Beta3 v1.0.0-beta02**

#### Steps to reproduce

The below method should return active Dataflow jobs located in a region specified in `location` parameter, but actually it returns jobs from `us-central1` region regardless of the parameter value.

```csharp
using Google.Cloud.Dataflow.V1Beta3;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using System;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;

namespace Abracadabra.Api.Controllers
{
    internal class GcpController : Controller
    {
        public async Task<IActionResult> GetDataflowJobsAsync(string project, string location, CancellationToken cancellationToken)
        {
            try
            {
                var client = await JobsV1Beta3Client.CreateAsync(cancellationToken);
                var request = new ListJobsRequest
                {
                    ProjectId = project,
                    Filter = ListJobsRequest.Types.Filter.Active,
                    Location = location
                };
                var jobs = client.ListJobsAsync(request);
                var result = await jobs.ToListAsync(cancellationToken);

                return Ok(result);
            }
            catch (Exception e)
            {
                return StatusCode(StatusCodes.Status500InternalServerError, e);
            }
        }
    }
}
```

**Comment by amanda-tarafa** [comment]
I'm looking into now. I'll report back here when I know more.

**Comment by amanda-tarafa** [comment]
According to the [API reference docs](https://cloud.google.com/dataflow/docs/reference/rpc/google.dataflow.v1beta3#jobsv1beta3) (scroll down to ListJobs()): 

> To list the jobs of a project in a region, we recommend using projects.locations.jobs.list with a regional endpoint. To list the all jobs across all regions, use projects.jobs.aggregated. Using projects.jobs.list is not recommended, as you can only get the list of jobs that are running in us-central1.

So the result you are seeing by using `JobsV1Beta3Client.ListJobsAsync` is expected. You may use `JobsV1Beta3Client.AggregatedListJobsAsync` to obtain jobs in all locations. As for the other part of the comment, I will ask internally, as I think there's no `projects.locations.jobs.list` functionality exposed at the moment by the gRPC Dataflow API. I'll get back here as I know more.

(Notice that the library is simply exposing, in and idiomatic manner, the functionality that the API itself exposes, so, if the API is not currently exposing a way to list Jobs per Location, then there's not much we can do on the library side. Still, I'll get clarity on this behaviour, I'll admit the API documentation is confusing).

**Comment by Rosti81** [comment]
Ah, sorry, this error is caused by the line

```csharp
var result = await jobs.ToListAsync(cancellationToken);
```

I'll check further and update you.

**Comment by amanda-tarafa** [comment]
Yes, it is the service failing nonetheless. When you execute `var jobs = client.ListJobsAsync(request);` (or the `Aggregated` one) this just creates a sequence, but makes no server calls. It's when you start iterating the sequence, in your case with `var result = await jobs.ToListAsync(cancellationToken);` that the server calls are performed as needed.

**Comment by amanda-tarafa** [comment]
Marking as blocked as we are in talks with the API team. For now, the best answer we have is for you to use  [Google.Apis.Dataflow.v1b3](https://www.nuget.org/packages/Google.Apis.Dataflow.v1b3/). I'll update here as soon as I have more information.

**Comment by meredithslota** [comment]
We've identified similar issues in other repos and have escalated this to the API service team since it's not a language-specific library issue. Here are the related issues: 

- https://github.com/googleapis/python-dataflow-client/issues/31: ListJobsRequest is not using the 'location' filter #31
- https://github.com/googleapis/python-dataflow-client/issues/25: Support for locations in the launch_template method #25
- https://github.com/googleapis/nodejs-dataflow/issues/19: The location option of launchFlexTemplate is not working #19
- https://github.com/googleapis/java-dataflow/issues/64: Dataflow: The location field of LaunchFlexTemplateRequest building is not working #64
- https://github.com/googleapis/google-cloud-dotnet/issues/7038: ListJobsAsync() returns only us-central1 jobs #7038 (context from issue is that the `location` parameter isn't working)

Marking this as a P1 bug w/ `external` label.

**Comment by bbhoss** [comment]
I couldn't find an existing issue in the Google Issue Tracker so I created one [here](https://issuetracker.google.com/issues/208105329). I suggest starring this issue to anyone else that runs across this problem, hopefully it will get some attention from Google soon.

**Comment by amanda-tarafa** [comment]
@meredithslota I'm assigning to you as I know you are talking to the API team. If there's anything we can do library side, feel free to reassing back to men.

**Comment by meredithslota** [comment]
The PR to address this (at least partially) has been merged: https://github.com/googleapis/google-cloud-dotnet/pull/7913 — can folks in this thread let me know if this resolves your issue?

**Comment by jskeet** [comment]
@meredithslota: Unfortunately it doesn't. I'll ping you internally with details.

**Comment by jskeet** [comment]
@Rosti81: This should now be fixed in a combination of version 1.0.0-beta03 of the library, and some internal changes. Note that the `Aggreate` RPCs still do *not* work and are being looked into separately - but the other RPCs should now work appropriately. Please let me know if you have run into any more problems.
